### DISCUSSION TITLE
BottomSheet Behavior

### DISCUSSION THREAD

**Discussion by avukalov** [VladislavAntonyuk/MauiSamples#339] [Q&A] [answered]
Hello Vladislav, 

I tried your BottomSheet example, and it works great. Is it possible to achieve the same behavior as shown in the Material documentation? You can find example [here](https://m2.material.io/components/sheets-bottom#behavior). If it is possible, could you provide me with some guidance?

Thank you!

**Comment by VladislavAntonyuk** [comment, marked as answer]
Hello Antonio, take a look at bottomSheetDialog.Behavior. In sample I use FitToContents. You can disable it to allow extending the BottomSheet.

**Reply by avukalov** [reply]
Thank you Vladislav.
 
Is there any documentation available on how to use platform-specific code? For example, your ShowBottomSheet extension method contains:

`new BottomSheetDialog(Platform.CurrentActivity?.Window?.DecorView.FindViewById(Android.Resource.Id.Content)?.Context`

where BottomSheetDialog constructor takes an Android.Content.Context. 

How can I determine how to access that context and similar things? I tried to find a parallel in the Java Google Android documentation, but I'm not sure how it is implemented in C#.

I'm very new to mobile development, and I don't know where to look.

Do I need to learn Android and iOS development before using MAUI if I want to build a fully-featured, well-designed mobile app?

**Reply by VladislavAntonyuk** [reply]
There are a lot of native controls for Android and iOS, you won't learn all of them and their API. The benefit of cross-platform frameworks is that you don't need to know all details of each platform, at least at the beginning level.
I suggest you start looking at .NET MAUI docs.

.NET Android SDK is a port of Java Google Android, so methods and controls are the same as in Java Google Android documentation.
